conn_go18.go

```diff
@@ -27,11 +27,14 @@ func (c *conn) Ping(ctx context.Context) error {
 		c.cancel = nil
 	}()
 	if err != nil {
-		return driver.ErrBadConn
+		return err
 	}
 	resp, err := ioutil.ReadAll(respBody)
-	if err != nil || len(resp) != 4 || !strings.HasPrefix(string(resp), "Ok.") {
-		return driver.ErrBadConn
+	if err != nil {
+		return err
+	}
+	if len(resp) != 4 || !strings.HasPrefix(string(resp), "Ok.") {
```

Comment:
Should we get actual response here? Like `user '<user>' not allowed` or smthng

Reply:
The response, in this case, will be an actual answer from ClickHouse. I'm not sure that it will be actual "user ... is not allowed", and could be a large string. But, yeah, we can return here `errors.New(resp)`, if it does not contain "Ok."

Reply:
From another point, this message could potentially have sensitive info, and throwing it back to the client will mean putting it to the application logs.

With current changes, the developer will see only that the response from ClickHouse is not Okay.